Dockerfile.template

```diff
@@ -1,18 +1,52 @@
-FROM resin/%%RESIN_MACHINE_NAME%%-debian:stretch
+FROM balenalib/%%BALENA_MACHINE_NAME%%-debian:buster-build as build
 
-ENV INITSYSTEM on
+WORKDIR /usr/src/balena-haskell-hello-world
+SHELL ["/bin/bash", "-c"] 
 
-# note: stackage resolver version should correspond to ghc version
-ARG ghc_version=8.0.1-17+rpi1
+# install ghcup
+RUN curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | BOOTSTRAP_HASKELL_NONINTERACTIVE=1 BOOTSTRAP_HASKELL_MINIMAL=1 bash
 
-RUN apt-get -q update && apt-get install -yq --no-install-recommends \
-	ghc=$ghc_version cabal-install haskell-stack netbase ca-certificates git \
-	&& apt-get clean && rm -rf /var/lib/apt/lists/*
+ARG GHC=8.10.7
+ARG stack=2.7.5
 
-WORKDIR /usr/src/app
+# # Add ghcup to PATH
+ENV PATH=${PATH}:/root/.local/bin
+ENV PATH=${PATH}:/root/.ghcup/bin
+ENV PATH=${PATH}:/usr/src/balena-haskell-hello-world/bin/
 
-COPY . /usr/src/app
+# # install GHC and cabal
+RUN \
+	install_packages llvm libnuma-dev && \
+	ghcup install ghc ${GHC} && ghcup set ghc ${GHC} && \
+	ghcup install stack ${stack} && ghcup set stack ${stack} 
 
-RUN stack setup && stack build
+COPY log/ log/
+COPY *.cabal . 
+COPY *.yaml . 
 
-CMD stack exec resin-haskell-hello-world
+RUN stack build --only-dependencies
+
+# COPY .ghci . 
+COPY src/ src/
+COPY views/ views/
+
+# RUN install_packages cabal-install file llvm && cabal update && cabal install cabal-install && cabal user-config update 
+RUN stack install --copy-bins
+# RUN bash -c "echo PATH="$HOME/.local/bin:$PATH" >> $HOME/.bashrc"
+# RUN install_packages haskell-stack
+# CMD [ "balena-haskell-hello-world", "-p", "80" ] 
+# RUN file ~/.local/bin/balena-haskell-hello-world
+
+# RUN cabal build --only-dependencies -j4
+# Add and Install Application Code
+# RUN cabal install
+
+FROM balenalib/%%BALENA_MACHINE_NAME%%-debian:buster-run 
+
+WORKDIR /usr/
+
+COPY --from=build /usr/src/balena-haskell-hello-world/bin/ src/
+COPY views/ views/
+
+# CMD [ "./src/Main" ] 
+CMD [ "tail", "-f", "/dev/null" ] 
```

Comment:
Is this a `sleep infinity` equivalent? I'm not familiar with it, but I've wondered if there should be a canonical "terminate and stay resident" command to avoid these confusing hacks. `sleep infinity` is the most obvious one I've seen.

Reply:
It is indeed.